x/ccv/provider/types/keys_test.go

```diff
@@ -124,43 +130,75 @@ func TestChainIdAndVscIdAndParse(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		key := chainIdAndVscIdKey(test.prefix, test.chainID, test.vscID)
+		key := types.ChainIdAndVscIdKey(test.prefix, test.chainID, test.vscID)
 		require.NotEmpty(t, key)
 		// Expected bytes = prefix + chainID length + chainID + vscId bytes
 		expectedLen := 1 + 8 + len(test.chainID) + 8
 		require.Equal(t, expectedLen, len(key))
-		parsedID, parsedVscID, err := parseChainIdAndVscIdKey(test.prefix, key)
+		parsedID, parsedVscID, err := types.ParseChainIdAndVscIdKey(test.prefix, key)
 		require.Equal(t, test.chainID, parsedID)
 		require.Equal(t, test.vscID, parsedVscID)
 		require.NoError(t, err)
 	}
 }
 
+// Tests the construction and parsing of ChainIdAndConsAddr keys
+func TestChainIdAndConsAddrAndParse(t *testing.T) {
+	pubKey1, err := testkeeper.GenPubKey()
+	require.NoError(t, err)
+	pubKey2, err := testkeeper.GenPubKey()
+	require.NoError(t, err)
+	pubKey3, err := testkeeper.GenPubKey()
+	require.NoError(t, err)
+
+	tests := []struct {
+		prefix  byte
+		chainID string
+		addr    sdk.ConsAddress
+	}{
+		{prefix: 0x01, chainID: "1", addr: sdk.ConsAddress(pubKey1.Address())},
+		{prefix: 0x02, chainID: "some other ID", addr: sdk.ConsAddress(pubKey2.Address())},
+		{prefix: 0x03, chainID: "some other other chain ID", addr: sdk.ConsAddress(pubKey3.Address())},
+	}
+
+	for _, test := range tests {
+		key := types.ChainIdAndConsAddrKey(test.prefix, test.chainID, test.addr)
+		require.NotEmpty(t, key)
+		// Expected bytes = prefix + chainID length + chainID + consAddr bytes
+		expectedLen := 1 + 8 + len(test.chainID) + len(test.addr)
+		require.Equal(t, expectedLen, len(key))
+		parsedID, parsedConsAddr, err := types.ParseChainIdAndConsAddrKey(test.prefix, key)
+		require.Equal(t, test.chainID, parsedID)
+		require.Equal(t, test.addr, parsedConsAddr)
+		require.NoError(t, err)
+	}
+}
+
```

Comment:
We could replace the custom <chainID, consAddr> pair implementation with a proto struct which we works, and we wouldn't have to test. Just a thought.

Reply:
Can you use a proto struct as a key in the SDK store? If yes, this will be useful to a lot of code (not just here). We can open a refactoring issue.

Reply:
Well of course you have to convert to bytes first. But the byte conversion is trivial and done for you without mistakes. We are just writing our own conversions everywhere which have caused bugs e.g

https://github.com/cosmos/interchain-security/issues/186

Reply:
It would be cool if the sdk could do the .bytes() call, or whatever, for you.

Reply:
I'm resolving this conversation as it is outside the scope of this PR.